Source/Entities/EntityInstanceBase.generated.cst

```diff
@@ -120,6 +127,11 @@ namespace <%=NameSpace%>
         <% } %>
         private <%=structName%> entityData;
 
+        /// <summary>
+        /// Plain object variable to place locks on when accessing _originalData.
+        /// </summary>
+        private object _originalDataLock = new object();

```

Comment:
I noticed we have some different patterns for locks, one was using volatile. https://stackoverflow.com/questions/11523003/should-a-lock-variable-be-declared-volatile (these should all be readonly).